pyfar/dsp/average.py

```diff
@@ -0,0 +1,102 @@
+def average(signal, average_mode='time', phase_copy=None,
+            weights=None):
+    """
+    Can be used to average impulse responses in different ways.
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the signal class
+    average_mode: string
+        'time' - averages in time domain
+        'complex' - averages the complex spectra
+        'abs' - averages the magnitude spectra
+        'power' - averages the power spectra
+        'log_magnitude' - averages the log magnitude spectra
+        The default is 'time'
+    phase_copy: vector
+        indicates signal channel from which phase is to be coppied
+        to the averaged signal
+        None - ignores the phase. Results in zero phase
+        The default is None
+    weights: array
+        array that gives weights for averaging the data
+        The default is None
+    Returns
+    --------
+    averaged_signal: Signal
+        averaged impulse responses
+    """
+
+    # check input
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal')
+
+    # check if 'weights' is right size
+
+    # average the data
+
+    if average_mode == 'time':
```

Comment:
Calling np.mean inside the if/elif statements is prone to errors. It would be better to only get the data here (time, abs, complex, ...) and apply the mean afterwards.